### PR TITLE
feat: auto content-type on blob creation

### PR DESCRIPTION
**Pull request by dmitry-fa**

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java [hunk 1]
```diff
@@ -372,6 +374,12 @@ public StorageObject create(
     }
   }
 
+  private static String detectContentType(StorageObject object) {
+    return firstNonNull(
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:378]: nit : I think you will need to test failed scenario as well.

**Review by athakor — APPROVED** [review]
LGTM with small nits

**Review by athakor — COMMENTED** [review]
Also, your` CI` build gets failed with following error, you will need to take care of it.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project google-cloud-storage: Compilation failure: Compilation failure: 
[ERROR] /home/runner/work/java-storage/java-storage/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:[84,24] package javax.activation does not exist
[ERROR] /home/runner/work/java-storage/java-storage/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:[102,24] cannot find symbol
[ERROR]   symbol:   class MimetypesFileTypeMap
[ERROR]   location: class com.google.cloud.storage.spi.v1.HttpStorageRpc
[ERROR] /home/runner/work/java-storage/java-storage/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:[102,75] cannot find symbol
[ERROR]   symbol:   class MimetypesFileTypeMap
[ERROR]   location: class com.google.cloud.storage.spi.v1.HttpStorageRpc
```

File: google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:378]: What do you mean?
There are two scenarios: the content type is set and not set. Both cases are covered by tests:

Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:378]: ya that's true but  what will happen if storage object is null??

Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:378]: NullPointerException will be thrown. 
BTW, other methods do not check storage object for null:
```
              .insert(
                  storageObject.getBucket(), <-- NPE will be thrown here
                  storageObject,
                  new InputStreamContent(detectContentType(storageObject), content));

```

Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:378]: Agreed, It's up to you, It's just suggestion.

**Review by pmakani — APPROVED** [review]

File: google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java:378]: Thanks

**Comment by dmitry-fa** [comment]
@athakor 
> Also, your CI build gets failed with following error, you will need to take care of it.

Thanks for pointing that. I added the dependency to work with java 11+
Now all the checks pass.

**Pull request closed without merging**